### PR TITLE
Fix Businessinsider & SfChronicle & Economist & BizJournals

### PR DESCRIPTION
**Pull request by magnolia1234**
Fix Businessinsider
General javascript-block blurs and hides pictures/videos.
New option to block external scripts on referring (enabled) domain.
Fixes issue #404
PS site stays enabled, but reset in options at next save (because of update in description).

Fix SfChronicle (reload loop)
Exclusion from localstorage.clear (contentScript.js).
Fixes issue #406

Fix Economist (only block external paywall-script)
PS site stays enabled, but reset in options at next save (because of update in description).

Fix BizJournals (no more Javascript-blocked necessary)
Fixes issue #358
PS site stays enabled, but reset in options at next save (because of update in description).

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -297,22 +299,36 @@ chrome.webRequest.onBeforeRequest.addListener(function(details) {
...
+  var header_referer = '';
+  for (var n in requestHeaders){
+	  if (requestHeaders[n].name.toLowerCase() == 'referer'){
+		  header_referer = requestHeaders[n].value;
+		  continue;
+	  }
+  }
+  
+  // check for blocked regular expression: domain enabled, match regex, block on an internal or external regex
+  for (var domain in blockedRegexes){
```
Comment (iamadamdev) [inline background.js]: Missing space before {

File: background.js [hunk 2]
```diff
@@ -297,22 +299,36 @@ chrome.webRequest.onBeforeRequest.addListener(function(details) {
...
   ["blocking"]
 );
 
 chrome.webRequest.onBeforeSendHeaders.addListener(function(details) {
-  if (!isSiteEnabled(details)) {
-    return;
+  var requestHeaders = details.requestHeaders;
+
+  var header_referer = '';
+  for (var n in requestHeaders){
```
Comment (iamadamdev) [inline background.js]: Missing space before {

(let's start enforcing styling standards)

Otherwise, great PR

File: background.js [hunk 3]
```diff
@@ -297,22 +299,36 @@ chrome.webRequest.onBeforeRequest.addListener(function(details) {
...
+	  if (requestHeaders[n].name.toLowerCase() == 'referer'){
```
Comment (iamadamdev) [inline background.js]: Missing space before {

File: background.js [hunk 4]
```diff
@@ -297,22 +299,36 @@ chrome.webRequest.onBeforeRequest.addListener(function(details) {
...
+		  header_referer = requestHeaders[n].value;
+		  continue;
+	  }
+  }
+  
+  // check for blocked regular expression: domain enabled, match regex, block on an internal or external regex
+  for (var domain in blockedRegexes){
+	  if (isSiteEnabled({url: '.'+ domain}) && details.url.match(blockedRegexes[domain])){
```
Comment (iamadamdev) [inline background.js]: Missing space before {

File: background.js [hunk 5]
```diff
@@ -297,22 +299,36 @@ chrome.webRequest.onBeforeRequest.addListener(function(details) {
...
+			if (details.url.indexOf(domain) !== -1 || header_referer.indexOf(domain) !== -1){
```
Comment (iamadamdev) [inline background.js]: Missing space before {

**Comment by iamadamdev** [comment]
I'll fix the styling issues on master

**Comment by willashley23** [comment]
I was just experiencing this bug a minute before you merged the PR in and tried test driving the PR myself, and even redownloaded the latest zip of master, but I think this might have broken it for SFChronicle, no more reload loop, but now I still get the adblock banner.

image

**Comment by iamadamdev** [comment]
Do you have uBlock Origin installed? I just opened 15+ articles and didn't get any paywall.

File: background.js (hunk 2 above)
Comment (magnolia1234) [reply background.js]: No space after for-loop? ;-)

**Comment by magnolia1234** [comment]
This anti-adblocker banner is not related to this extension though.
I don't get it with or without uBlock Origin enabled.

**Pull request merged**